### DISCUSSION TITLE
tvOS/Siri remote support?

### DISCUSSION THREAD

**Discussion by leonchabbey** [software-mansion/react-native-gesture-handler#2181] [Q&A]
I'm using https://github.com/react-native-tvos/react-native-tvos and i'm trying to get gestures such as Pan and Tap from the Apple TV remote but nothing happens (no callback triggered whatsoever).

This library is compatible with tvOS but there is no mention anywhere in the docs about how much of this library's features also work on Apple TV (and Android TV).

Did anyone successfully make it work on Apple TV / Android TV ?

Short example code snippet:
```typescript
<TapGestureHandler
  onHandlerStateChange={(state) => { // never triggered
    console.log('state changed', state);
  }}
>
  <TouchableOpacity />
</TapGestureHandler>
```

**Comment by jonybekov** [comment]
You should use TVEventControl and useTVEventHandler to capture pan, tap and etc.

See this: https://github.com/react-native-tvos/react-native-tvos/pull/295

**Comment by tamish-max** [comment]
For Apple TV I would not rely on RNGH as the primary input layer.

The Siri remote is much closer to a **TV event / focus** device than a touch screen, so the more reliable approach is usually:

- `useTVEventHandler` / `TVEventHandler` from `react-native-tvos`
- focus-based props (`onFocus`, `onBlur`, `hasTVPreferredFocus`)
- `TVEventControl` if you need to tune menu/pan behavior

RNGH is great for touch gestures on iOS/Android, but tvOS remote input is a different model. So if Pan/Tap callbacks are not firing, that is not surprising.
